### PR TITLE
Sp enhancement schema and create

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**

### CODE REVIEW

File: docs/resources/storage_pool.md [hunk 1]
```diff
@@ -36,16 +36,35 @@ This resource can be used to manage Storage Pools on a PowerFlex array.
...
-  protection_domain_id = "4eeb304600000000"
-  # protection_domain_name = "domain1"
+  #protection_domain_id = "4eeb304600000000"
+  protection_domain_name = "domain1"
   media_type  = "HDD"
-  use_rmcache = true
-  use_rfcache = false
+  use_rmcache = false
+  use_rfcache = true
+  #replication_journal_capacity = 34
```
Comment (shenda1) [inline docs/resources/storage_pool.md:49]: Do we need to comment this parameter?

Comment (shenda1) [reply docs/resources/storage_pool.md:49]: @AnikaAgiwal2711 Its fine.

File: powerflex/storagepool_resource.go [hunk 2]
```diff
@@ -53,6 +73,130 @@ func (r *storagepoolResource) Configure(_ context.Context, req resource.Configur
 	r.client = req.ProviderData.(*goscaleio.Client)
 }
 
+func (r *storagepoolResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
```
[the remark this answers is not in the document]
Comment (AnikaAgiwal2711) [reply powerflex/storagepool_resource.go:76]: Yeah Sure. I can document combination in notes and will share that notes to all.

[the remark this answers is not in the document]
Comment (Krishnan-Priyanshu) [reply powerflex/storagepool_resource.go:76]: all the documents and note has been shared.

**Review by shenda1 — APPROVED** [review]

**Review by Krishnan-Priyanshu — APPROVED** [review]
LGTM

**Pull request merged**